### ISSUE TITLE
Excluding files doesn't work when using STDIN with a filename specified

### ISSUE THREAD

**Issue by rkit** [squizlabs/PHP_CodeSniffer#1208]
Excluding files doesn't work when using `stdin`

Config:

```xml
<?xml version="1.0"?>
<ruleset name="Custom Standard">
    <description>A custom coding standard</description>

    <exclude-pattern>*/views/*</exclude-pattern>
    <rule ref="PSR2" />
</ruleset>
```

Command:
```sh
cat /test/views/index.php | phpcs --standard="/test/phpcs.xml" --stdin-path="/test/views/index.php"
```

And as a result, we got errors.

Is it a known bug? I would like to fix it

```
PHP_CodeSniffer version 2.7.0 (stable) by Squiz (http://www.squiz.net)
```

**Comment by gsherwood** [comment]
PHPCS has always assumed that if you are passing content directly to it, you want it checked no matter what the exclude rules are. I have changed this in 3.0 (due to request #733), but it is a serious backwards compatibility break so it wont be changed in the 2.x version.

If you'd like to try out the current 3.0 RC release, you should find that it works as you expect.

**Comment by rkit** [comment]
> due to request #733

Yes, I use this [linter](https://github.com/AtomLinter/linter-phpcs) for the Atom

> If you'd like to try out the current 3.0 RC release, you should find that it works as you expect.

I tried with 3.0.0RC1 and the result is the same, I have got errors.

```
PHP_CodeSniffer version 3.0.0RC1 (beta) by Squiz (http://www.squiz.net)
```

**Comment by gsherwood** [comment]
> I tried with 3.0.0RC1 and the result is the same, I have got errors.

Sorry, it was working and I've realised it broke when the stdin-path feature was merged in from master. I've fixed it again now.

**Comment by rkit** [comment]
Thanks!
